From a7eaefe6ec5fff94f9dd152c279809c95be3946c Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Fri, 20 Nov 2015 13:50:20 +0800
Subject: [PATCH 0678/5242] MLK-11881 video: fbdev: mxc ipuv3 fb: Correct
 ->prefetch in ioctrl SET_PREFETCH

commit  afdc105b87c11ed0ffc9e0ccca50a6d89e568019 from
https://source.codeaurora.org/external/imx/linux-imx.git

When a framebuffer is blanked, the user may call the MXCFB_SET_PREFETCH ioctrl
to enable or disable the prefetch engine.  After sanity check, the cached
->prefetch should reflect the user's intention for next unblank operation
any way instead of checking ->cur_prefetch first.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_ipuv3_fb.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
index 3af13e6..eb9b2f2 100644
--- a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
@@ -2047,15 +2047,15 @@ static int mxcfb_ioctl(struct fb_info *fbi, unsigned int cmd, unsigned long arg)
 				}
 			}
 
-			if (mxc_fbi->cur_prefetch == !!enable)
-				break;
-
 			retval = mxcfb_check_var(&fbi->var, fbi);
 			if (retval)
 				break;
 
 			mxc_fbi->prefetch = !!enable;
 
+			if (mxc_fbi->cur_prefetch == mxc_fbi->prefetch)
+				break;
+
 			fbi->var.activate = (fbi->var.activate & ~FB_ACTIVATE_MASK) |
 						FB_ACTIVATE_NOW | FB_ACTIVATE_FORCE;
 			retval = mxcfb_set_par(fbi);
-- 
1.7.9.5

